spanner/pdml.go

```diff
@@ -114,7 +118,13 @@ func executePdml(ctx context.Context, sh *sessionHandle, req *sppb.ExecuteSqlReq
 	req.Transaction = &sppb.TransactionSelector{
 		Selector: &sppb.TransactionSelector_Id{Id: res.Id},
 	}
-	resultSet, err := sh.getClient().ExecuteSql(contextWithOutgoingMetadata(ctx, sh.getMetadata()), req)
+	resultSet, err := sh.getClient().ExecuteSql(contextWithOutgoingMetadata(ctx, sh.getMetadata()), req, gax.WithGRPCOptions(grpc.Header(&md)))
+	if GFELatencyMetricsEnabled && md != nil && sh.session.pool != nil {
```

Comment:
Also here: Check the error from the `ExecuteSql` method before trying to extract the metadata to prevent us from returning a potentially confusing error to the user.